File: frontend/src/types/order.ts
export interface OrderItem {
  item_id: number
  goods_id: number
  goods_name?: string
  quantity: number
  price: number
}

export interface OrderAddress {
  address_name?: string
  address_phone?: string
  address_province?: string
  address_city?: string
  address_district?: string
  address_detail?: string
}

export interface Order {
  order_id: number
  user_id: number
  total_price: number
  status: string
  create_time: string
  address_name?: string
  address_phone?: string
  address_province?: string
  address_city?: string
  address_district?: string
  address_detail?: string
}

export interface OrderDetail extends Order {
  items: OrderItem[]
}

export interface OrderListResponse {
  items: Order[]
}

// ==================== 退换货相关类型 ====================

export interface OrderReturn {
  return_id: number
  order_id: number
  user_id: number
  type: string
  status: string
  reason: string
  images?: string
  remark?: string
  approve_remark?: string
  create_time: string
  update_time: string
}

export interface OrderReturnCreate {
  order_id: number
  type: string
  reason: string
  images?: string
  remark?: string
}

export interface OrderReturnListResponse {
  items: OrderReturn[]
  total: number
}

export interface CheckoutRequest {
  goods_id: number
  quantity: number
  address_id: number
}

export interface CartCheckoutRequest {
  cart_ids: number[]
  address_id: number
}

File: shop-service/app/models/order.py
from sqlalchemy import Column, Integer, String, Numeric, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database import Base


class Order(Base):
    __tablename__ = "orders"

    order_id = Column(Integer, primary_key=True, autoincrement=True, index=True)
    user_id = Column(Integer, nullable=False, index=True)
    total_price = Column(Numeric(10, 2), nullable=False)
    status = Column(String(20), default="pending", nullable=False)
    create_time = Column(DateTime(timezone=True), server_default=func.now())

    # 收货地址（快照保存）
    address_name = Column(String(50), nullable=True, comment="收货人姓名")
    address_phone = Column(String(20), nullable=True, comment="收货人电话")
    address_province = Column(String(50), nullable=True, comment="省份")
    address_city = Column(String(50), nullable=True, comment="城市")
    address_district = Column(String(50), nullable=True, comment="区县")
    address_detail = Column(String(200), nullable=True, comment="详细地址")

    items = relationship("OrderItem", back_populates="order")


class OrderItem(Base):
    __tablename__ = "order_items"

    item_id = Column(Integer, primary_key=True, autoincrement=True, index=True)
    order_id = Column(Integer, ForeignKey("orders.order_id"), nullable=False)
    goods_id = Column(Integer, nullable=False)
    quantity = Column(Integer, nullable=False)
    price = Column(Numeric(10, 2), nullable=False)

    order = relationship("Order", back_populates="items")
